papas/detectors/clic/src/ClicTracker.cpp

```diff
@@ -48,30 +48,29 @@ double ClicTracker::resolution(const Track& track) const {
   double pt = track.p3().Pt();
   // matching the resmap defined above.
   double theta = fabs(track.theta()) * 180 / M_PI;
-  for (const auto& v : m_resMap) { 
+  for (const auto& v : m_resMap) {
     if (theta < v.first) return sigmaPtOverPt2(v.second.first, v.second.second, pt) * pt;
   }
   throw "tracker resolution not found";
   return 0;
 }
 
-  double ClicTracker::particleResolution(const Particle& ptc) const {
-    //not very nice due to particle being fed through into a track parameter in python
-    //will do for now
-    double pt = ptc.p3().Pt();
-    // matching the resmap defined above.
-    double theta = fabs(ptc.theta()) * 180 / M_PI;
-    for (const auto& v : m_resMap) {
-      if (theta < v.first) return sigmaPtOverPt2(v.second.first, v.second.second, pt) * pt;
-    }
-    throw "tracker particle resolution not found";
-    return 0;
-  }
-  
-  bool ClicTracker::electronAcceptance(double mag, const Track& track) const {
-    if ( track.p3().Pt() > mag && fabs(track.theta()) < m_thetaMax)
-      return rootrandom::Random::uniform(0, 1) > 0.95;
-    return false;
+double ClicTracker::particleResolution(const Particle& ptc) const {
+  // not very nice due to particle being fed through into a track parameter in python
+  // will do for now
+  double pt = ptc.p3().Pt();
+  // matching the resmap defined above.
+  double theta = fabs(ptc.theta()) * 180 / M_PI;
+  for (const auto& v : m_resMap) {
+    if (theta < v.first) return sigmaPtOverPt2(v.second.first, v.second.second, pt) * pt;
   }
-  
-}// end namespace papas
+  throw "tracker particle resolution not found";
```

Comment:
Naive question: shouldn't that be a PAPASLOG_ERROR?
Maybe review the use of throw to see which ones should be a PAPASLOG_ERROR

Reply:
I think this is OK. The idea is that papas can throw during an event and then  this should be caught and the event discarded ie there needs to be somewhere that centrally manages any issues in processing a specific event. So I judged that it is OK to throw to the centralized area and then issue a formal error such as PAPASLOG_ERROR  at this point.  This means we have most of the calls to PAPASLOG_ERROR just after the throw is caught. In fact in gaudi once the throw is caught the error can go direct to its logger. I hope that makes sense :-). It seems to work.